format_spec.md

```diff
@@ -0,0 +1,339 @@
+- `alwaysAllow` - This Project can be used in a modpack without asking for permission.
+- `requiresPermission` - Before included in a modpack, permission must be granted from the author.
+- `neverAllow` - This Project can never be used in a modpack.
+
+---
+
+### `links` (optional)
+
+This field MUST be stored as an array. It contains a collection of links relating to the Project. Each Link Object stored inside this arrow MUST contain the following fields:
+
+##### `rel`
+
+This field MUST be stored as an array OR a String. If stored as an array, it MUST contain Strings. This field contains information about what the link is. Valid standardized values are:
+
+- `homepage` - The official homepage of a Project
+- `documentation` - Documentation of a Project.
+- `wiki` - Wiki information about a Project.
+- `scm` - The source code of a Project, for example a GitHub URL.
+- `issues` - An issue tracker where bugs and feature requests can be filed.
+- `community` - A forum or other online discussion platform where users can discuss the Project.
+
+Other values MAY be used, such as `bitbucket` or `github`. However, it is RECOMMENDED to use generic terminology instead of specific names. If you are unsure whether a value COULD become standardized, please prefix it with `x-`.
+
+##### `name`
+
+A human-readable name of the link. For example, "Source Code" or "Project Website".
+
+##### `uri`
+
+The link, to be viewed in a web browser.
+
+---
+
+### `successor` and `predecessor` (optional)
+
+If an older or newer version of a Project exists, but has a separate ID, these fields may be used. These fields MUST be either an array or String representing the respective projects' IDs. If the field is an array, the Projects should be listed in chronological order from when they were first released.
+
+```
+"successor": [
+  "first-successor",
+  "second-successor
+]
+```
+
+In the example above, the project `first-successor` was released first. After the release of `first-successor`, the project `second-successor` was released.
+
+---
+
+## Version Information
+
+Projects can contain a `versions` array, which lists Version Objects.
+
+## Version Objects
+
+Version Objects contain fields relating to a certain version of a Project. Below are all the fields that a Version Object may contain.
+
+### `id`
+
+The ID of this version. There is no required format for how to format a version ID, other than that they CANNOT include spaces.
+
+---
+
+### `name`
+
+The human-readable name of this version. It is RECOMMENDED to follow Semantic Versioning, but it is not required. Spaces MAY be used in this name.
+
+---
+
+### `semver` (optional)
+
+An optional Semantic Versioning compatible version number. This field SHOULD ONLY be present if the version's name conforms to Semantic Versioning. This field SHOULD NOT be present if the name does not conform to SemVer.
+
+---
+
+### `releaseDate`
+
+The release date of this version. It MUST be stored as an ISO-8601 conforming String.
```

Comment:
Does it include time? Timezonr?

Reply:
ISO8601 can include time and timezone, so I'd think an ISO-8601 conforming String would include that.

Reply:
I was thinking to use UTC (e.g. `2020-08-05T12:00:00Z` is 12:00 UTC).

Reply:
I've done a non-trivial amount of work with dates and timezones and the key thing to determine how to store the data amounts to "is this a thing that happened/is happening. or a thing that will happen?"

If it is something that happened in the past, or a record of something happening (timestamp, event log) then UTC is probably the easiest and safest bet. If it is happening in the future, you need to log the actual desired timezone along with the ISO-8601 data. ISO-8601 actually does not attempt to store timezone data. There is an optional "Z" indicator which is supposed to indicate that the time is in UTC; in practicality, nearly all ISO-8601 datetimes are in UTC so most systems ignore the Z.

Some systems store the offset of the timezone along with the date, but that is unfortunately also insufficient. If someone wants to record an event occurring on 2024/06/04T08:30:00+05:00, but the local government decides to stop observing DST between storing the date and retrieving it, then an inconsistency is introduced.

Reply:
Since we are only recording things that happened, then, it's UTC.

The documentation should give an example value. xxxx-xx-xxTxx:xx:xxZ. No others. According to Wikipedia, 2020-08-05 is also an ISO-8601 conforming string. So is 2020-W32. So is --08-05. So is 2020-218. But we only want a particular format.

If the information is not known, should it be approximated? How much accuracy is needed?

Reply:
I'm working on modifying the documentation and providing an example value, `2020-01-01T12:00:00Z`.

If information about the time a version was released is unknown, it should default to 12:00:00 UTC. I don't know how much precision should go with the actual date, but at some point the value can default to `unknown`. The defaulting of release dates should fall to hosts, and launchers shouldn't have to generate default date values.